### PR TITLE
make rake proxy work in rails engines

### PR DESCRIPTION
**Pull request by y-yagi**
To avoid confusion, even Rails engines, I think that it may be able to run the rake task over the rails command. How about?

### CODE REVIEW

**Comment by maclover7** [comment]
cc @kaspth

File: railties/lib/rails/engine/rake_proxy.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          Rake.application.load_rakefile
+          @rake_tasks = Rake.application.tasks.select(&:comment)
+        end
+
+        def formatted_rake_tasks
+          rake_tasks.map { |t| [ t.name_with_args, t.comment ] }
+        end
+    end
+  end
+end
```
Comment (kaspth) [inline railties/lib/rails/engine/rake_proxy.rb]: Why can't we reuse the proxy from main Rails?

Comment (y-yagi) [inline railties/lib/rails/engine/rake_proxy.rb]: And I thought too much or reuse, but it is almost the same when the review. 
I'm sorry, and then modified to reuse the rake_proxy of Rails.

Comment (y-yagi) [inline railties/lib/rails/engine/rake_proxy.rb]: Updated.

File: railties/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+*   Make rake proxy work in Rails engines.
+
+    *Yuji Yaginuma*
+
```
Comment (kaspth) [inline railties/CHANGELOG.md]: Don't think this is CHANGELOG worthy, let's assume it worked in engines always.

File: railties/lib/rails/engine/commands_tasks.rb [hunk 3]
```diff
@@ -0,0 +1,116 @@
+require 'rails/commands/rake_proxy'
```
Comment (kaspth) [inline railties/lib/rails/engine/commands_tasks.rb]: Needs to require `rake` too, no?

**Comment by kaspth** [comment]
Please rebase and squash your commits too :grin:

File: railties/lib/rails/engine/commands_tasks.rb [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+        def rake_tasks
+          return @rake_tasks if defined?(@rake_tasks)
+
+          load_generators
+          Rake::TaskManager.record_task_metadata = true
+          Rake.application.init('rails')
+          Rake.application.load_rakefile
+          @rake_tasks = Rake.application.tasks.select(&:comment)
+        end
```
Comment (kaspth) [inline railties/lib/rails/engine/commands_tasks.rb]: Isn't `rake_tasks` included from the `Rails::RakeProxy`?

File: railties/lib/rails/engine/commands_tasks.rb (hunk 3 above)
Comment (y-yagi) [inline railties/lib/rails/engine/commands_tasks.rb]: Has require a `rake` in the `rails/commands/rake_proxy`, I think here that it is not need.

File: railties/lib/rails/engine/commands_tasks.rb (hunk 4 above)
Comment (y-yagi) [inline railties/lib/rails/engine/commands_tasks.rb]: `Rails::RakeProxy#rake_tasks` has become a prerequisite of the code that uses the Rails application, did not work for as it is the case of the Rails Engine, it was defined separately.

File: railties/CHANGELOG.md (hunk 2 above)
Comment (y-yagi) [inline railties/CHANGELOG.md]: Okay.

**Comment by y-yagi** [comment]
Thanks for reviewing!  Rebase and squash commits :)

**Comment by kaspth** [comment]
Thanks!

**Comment by y-yagi** [comment]
Thanks!

**Pull request merged**